### DISCUSSION TITLE
QR Control Bug? GP 11 record stops after about 1 minute

### DISCUSSION THREAD

**Discussion by stpfl123** [gopro/labs#410] [General]
[HERO11 Black]v2.10.70
QRControl 1.8.3

After updating my 2x  Gopro 11 to Goprolabs 2.10.70 and using QRControl 1.8.3 i am getting a bug after several QRControl commands. 
Then camera will stop record after 1 minute. The problem is with two cameras and different micro sd cards. Standard High Bitrate Setting. 
When i make a Hard Reset on Gopro the bug is gone but will come back aggain after several QRControl commads ?

**Comment by dnewman-gpsw** [comment]
What commands?   Some will cause that by design.  I do not expect there is a bug.

**Comment by stpfl123** [comment]
Wake up when power
ISO and Shutter Display on Front Display
Standard / 4k 8x7 / 50p / Wide / 10bit / High Bitrate / 4000K / Iso min 100 / iso max 800 / Sharpness Low / Colour Flat/ 
nothing else

**Comment by stpfl123** [comment]
> What commands? Some will cause that by design. I do not expect there is a bug.

which commands ?

**Comment by stpfl123** [comment]
first time a had this problem was on June 4th then full reset working for about 20 flights now today problem came back... 
Last half year no problems, same settings
same problem on two cameras with differnt SD cards. 
last update QRcontrol 27.05.2023 ?

**Comment by dnewman-gpsw** [comment]
Please share the actually command string, or QR Code that causes it.

**Comment by dnewman-gpsw** [comment]
Is it exactly one minute?

**Comment by dnewman-gpsw** [comment]
Also a metadata extraction using https://gopro.github.io/labs/control/metadata/ for one of the failed captures.

**Comment by stpfl123** [comment]
> Is it exactly one minute?

1 minute 4 seconds

**Comment by stpfl123** [comment]
> Please share the actually command string, or QR Code that causes it.

DEVC 
  DVID 1
  DVNM Global Settings
  VERS 8, 2, 2
  FMWR H22.01.02.10.70
  LINF LSU2060701401266
  CINF 104, 183, 23, 207, 14, 96, 189, 59, 223, 183, 213, 102, 224, 60, 34, 79
  CASN C3471325066261
  MINF HERO11 Black
  MUID 3474437992, 1002266638, 1725282271, 1327643872, 0, 0, 0, 0
  CMOD 12
  MTYP 0
  OREN U
  DZOM Y
  DZST 0
  SMTR N
  PRTN Y
  PTWB 4000K
  PTSH LOW
  PTCL FLAT
  EXPT AUTO
  PIMX 800
  PIMN 100
  PTEV 0.0
  RATE .empty.
  SROT 16.0000
  EISE N
  EISA N/A
  HCTL Off
  AUPT N
  APTO OFF
  AUDO AUTO
  BROD .empty.
  BRID 0
  PVUL F
  PRJT GPRO
  SOFF 0
  CLKS 1
  CDAT 0x00000000639F9C17
  SCTM 0
  PRNA 1
  PRNU 0
  SCAP N
  CDTM 0
  DUST NO_LIMIT
  VRES 3840, 3360
  VFPS 50000, 1000
  HSGT OFF
  BITR HIGH
  MMOD STEREO
  RAMP .empty.
  TZON 32767
  CLKC 0
  DZMX 1.4000
  CTRL Pro
  PWPR PERFORMANCE
  ORDP Y
  CLDP Y
  PIMD MANUAL
DEVC 
  DVID FOVL
  DVNM Large FOV
  ABSC 0.0000
  ZFOV 156.7126
  VFOV W
  MXCF x1
  MAPX 1.0000
  MYCF y1
  MAPY 1.0000
  PYCF r0r1r2r3r4r5r6
  POLY 0.0000, 2.1112, 0.1433, -1.0301, 0.6410, -0.0000, 0.0000
  ZMPL 0.7115
  ARUW 1.1429
  ARWA 1.1429
DEVC 
  DVID USRM
  DVNM User Metadata
  STRM 
    WAKE 2
    FRNT 1
    EXPS 1
    EXPX 60
    USRD 1
DEVC 
  DVID HLMT
  DVNM Highlights

**Comment by dnewman-gpsw** [comment]
You haven't shared the command yet. As you have exposure max EXPX set at 60, WAKE=2, and display the ISO/Shutter on the FRNT display, guessing these are **!M** settings, not **oM** settings?  While this aren't the cause either, all my naked cameras are configured similarly.  So I guessing this is a naked config?  Naked is more unreliable, do you have sufficient cooling?  Also I'm surprised as an FPV user you aren't using BITR to increase the bitrate.  While this might be counterintuitive !MBITR=150 might help.  The bitrate hack includes some enhanced monitoring of the SD Card performance, so it might help.   

As this is not a bug in general, I moving this to discussions.

**Comment by stpfl123** [comment]
i don't have the command anymore because i reset and deinstalled qrcontrol. i will share when i can reproduce the problem aggain. 
But i am still wondering what excactly causing the problem because this exact setup was running fine before...

**Reply by dnewman-gpsw** [reply]
Likely a naked camera over heating, but I'm not sure. Hardware does age, particularly SD cards. Use **!MDBGL=1** to enable more debug logging.  Then share the file MISC/qrlog.txt directly after the next capture failure.  The log might explain more.

**Reply by stpfl123** [reply]
One of the two camera has under 10 records so is very new. Same cooling since 1 year problem since 10 days
also sd cards are very new. it worked before !!!
I will log now

**Reply by stpfl123** [reply]
question: when i use more commands the mainboard gets hotter? so can there be a difference between one or less command between overheating or not ?

**Reply by dnewman-gpsw** [reply]
No. Commands are settings, same as using the touchscreen menus.  The !MEXPS=1 command is doing work, but that is not huge.

**Reply by dnewman-gpsw** [reply]
Are you scanning for each flight?  I don't that.  I use a boot command with WAKE=2, so configure a motion trigger, so the camera records only while in flight.  This is also help with thermals, as pressing record on a naked camera before the rotor are running, will be an opportunity to overheat a camera.

**Reply by stpfl123** [reply]
i would like to start record with imu detection but want to stop recording by myself to check if it was recorded.
I did't find the permanet command for this till know. maybe you can tell me ? :-)

**Reply by dnewman-gpsw** [reply]
Many ways to do this now.

Here is a boot command and combination trigger that uses a manual record stop.

!MBOOT="!Lspl"!SAVEspl=<r0>p80>a0.25mV+!S!R

!MBOOT="!Lspl" -  this part makes the camera look for a file MISC/spl and runs that on boot.

!SAVEspl=...  - this part saves whatever you want in the boot script.

Boot script:
```
<r0>p80>a0.25   - If(not_recording and sound > 80dB and acceleration > 0.25g)
    mV+!S       - then set mode Video and start capture
!R              - Repeat
```

I've add the sound pressure level test, as it prevent recording starts when you pick up the drone at the end of a flight.  So rotor (noise) + acceleration to start recording. 

![image](https://github.com/gopro/labs/assets/10856084/2168760f-5c1e-4fd2-be5c-697ad5d6d6cc)

**Reply by stpfl123** [reply]
Thanks !!!!

**Comment by stpfl123** [comment]
It happenend aggain. After many records without problems record stoppend after about 30 seconds...
Logfile: 
2022-12-18 23:01:55] Labs:Main Start 
[2022-12-18 23:01:55] Labs LCD ID:1 
[2022-12-18 23:01:55] OLED width:0 
[2022-12-18 23:01:55] OLED height:0 
[2022-12-18 23:01:55] Naked when 0:0 
[2022-12-18 23:01:55] Labs:Initialize 
[2022-12-18 23:01:55] display:Boot command:
!Lspl 
[2022-12-18 23:01:56] Boot command:!Lspl 
[2022-12-18 23:01:57] Use CMD:!Lspl 
[2022-12-18 23:01:57] ACTION:!Lspl 
[2022-12-18 23:01:58] Load Script:C:\MISC\spl 
[2022-12-18 23:01:58] ACTION:!SI6H5!R 
[2022-12-18 23:02:06] GP2 Temp.:55 
[2022-12-18 23:02:13] start capture:IMU 0.010g+1dps  threshold = 0.032g+15dps 
[2022-12-18 23:02:15] GP2 Temp.:57 
[2022-12-18 23:02:18] GP2 Temp.:61 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:20] capture stopped:user input or error 
[2022-12-18 23:02:21] trigger capture stopped:user input or error 
[2022-12-18 23:02:24] GP2 Temp.:58

**Comment by stpfl123** [comment]
QR Code Command 
https://drive.google.com/file/d/13UYNoRKer_vulm3K4IlR_q41X3JvmWe3/view?usp=sharing

**Comment by stpfl123** [comment]
DEVC 
  DVID 1
  DVNM Global Settings
  VERS 8, 2, 2
  FMWR H22.01.02.10.70
  LINF LSU2060701401266
  CINF 104, 183, 23, 207, 14, 96, 189, 59, 223, 183, 213, 102, 224, 60, 34, 79
  CASN C3471325066261
  MINF HERO11 Black
  MUID 3474437992, 1002266638, 1725282271, 1327643872, 0, 0, 0, 0
  CMOD 12
  MTYP 0
  OREN U
  DZOM Y
  DZST 0
  SMTR N
  PRTN Y
  PTWB 4000K
  PTSH LOW
  PTCL FLAT
  EXPT AUTO
  PIMX 800
  PIMN 100
  PTEV 0.0
  RATE .empty.
  SROT 16.0000
  EISE N
  EISA N/A
  HCTL Off
  AUPT N
  APTO OFF
  AUDO AUTO
  BROD .empty.
  BRID 0
  PVUL F
  PRJT GPRO
  SOFF 0
  CLKS 1
  CDAT 0x00000000639F9D12
  SCTM 0
  PRNA 1
  PRNU 0
  SCAP N
  CDTM 0
  DUST NO_LIMIT
  VRES 3840, 3360
  VFPS 50000, 1000
  HSGT OFF
  BITR HIGH
  MMOD STEREO
  RAMP .empty.
  TZON 32767
  CLKC 0
  DZMX 1.4000
  CTRL Pro
  PWPR PERFORMANCE
  ORDP Y
  CLDP Y
  PIMD MANUAL
DEVC 
  DVID FOVL
  DVNM Large FOV
  ABSC 0.0000
  ZFOV 156.7126
  VFOV W
  MXCF x1
  MAPX 1.0000
  MYCF y1
  MAPY 1.0000
  PYCF r0r1r2r3r4r5r6
  POLY 0.0000, 2.1112, 0.1433, -1.0301, 0.6410, -0.0000, 0.0000
  ZMPL 0.7115
  ARUW 1.1429
  ARWA 1.1429
DEVC 
  DVID USRM
  DVNM User Metadata
  STRM 
    EMPT 4
    DBGL 1
    FRNT 1
    EXPS 1
    WAKE 2
    BOOT !Lspl
    USRD 
DEVC 
  DVID HLMT
  DVNM Highlights

**Comment by stpfl123** [comment]
2second test after Error:  Front Display does not turn off after one minute anymore? 
3 th test without any QR Controll Command Front Display turn off after one minute aggain

**Comment by stpfl123** [comment]
[2022-12-18 23:01:56] Boot command:!Lspl 
[2022-12-18 23:01:56] Use CMD:!Lspl 
[2022-12-18 23:01:56] ACTION:!Lspl 
[2022-12-18 23:01:56] Failed to load Script:C:\MISC\spl 
[2022-12-18 23:02:00] GP2 Temp.:34 
[2022-12-18 23:02:03] GP2 Temp.:39 
[2022-12-18 23:02:06] GP2 Temp.:45 
[2022-12-18 23:02:10] GP2 Temp.:49 
[2022-12-18 23:02:14] GP2 Temp.:52 
[2022-12-18 23:02:19] GP2 Temp.:55 
[2022-12-18 23:02:25] GP2 Temp.:57 
[2022-12-18 23:02:30] GP2 Temp.:60 
[2022-12-18 23:02:36] GP2 Temp.:62 
[2022-12-18 23:02:42] GP2 Temp.:64 
[2022-12-18 23:02:49] GP2 Temp.:66 
[2022-12-18 23:02:57] GP2 Temp.:68 
[2022-12-18 23:03:08] GP2 Temp.:69 
[2022-12-18 23:03:16] GP2 Temp.:71 
[2022-12-18 23:03:27] GP2 Temp.:72 
[2022-12-18 23:03:38] GP2 Temp.:73 
[2022-12-18 23:03:47] GP2 Temp.:75 
[2022-12-18 23:03:59] GP2 Temp.:76 
[2022-12-18 23:04:05] wake reason:HEARTBEAT_TIMEOUT 
[2022-12-18 23:01:50] display:Features activated: 

Or sometimes it get stucked when i want to stop recording see above. what does heartbeat timeout mean?

**Comment by stpfl123** [comment]
Nr 3 Record Stop after 15 seconds  - with a different Sandisk Extreme Card v30 

[2022-12-18 23:01:54] Labs:Main Start 
[2022-12-18 23:01:54] Labs LCD ID:1 
[2022-12-18 23:01:54] OLED width:0 
[2022-12-18 23:01:54] OLED height:0 
[2022-12-18 23:01:54] Naked when 0:0 
[2022-12-18 23:01:54] Labs:Initialize 
[2022-12-18 23:01:54] display:Boot command:
!Lspl 
[2022-12-18 23:01:56] Boot command:!Lspl 
[2022-12-18 23:01:56] Use CMD:!Lspl 
[2022-12-18 23:01:56] ACTION:!Lspl 
[2022-12-18 23:01:56] Failed to load Script:C:\MISC\spl 
[2022-12-18 23:02:00] GP2 Temp.:35 
[2022-12-18 23:02:03] GP2 Temp.:40 
[2022-12-18 23:02:24] heartbeat:main qr loop is locked 
[2022-12-18 23:02:24] heartbeat cmd write: 
[2022-12-18 23:02:28] wake reason:RESET

**Comment by stpfl123** [comment]
DEVC 
  DVID 1
  DVNM Global Settings
  VERS 8, 2, 2
  FMWR H22.01.02.10.70
  LINF LSU2060701401266
  CINF 104, 183, 23, 207, 14, 96, 189, 59, 223, 183, 213, 102, 224, 60, 34, 79
  CASN C3471325066261
  MINF HERO11 Black
  MUID 3474437992, 1002266638, 1725282271, 1327643872, 0, 0, 0, 0
  CMOD 12
  MTYP 0
  OREN U
  DZOM Y
  DZST 0
  SMTR N
  PRTN Y
  PTWB 4000K
  PTSH LOW
  PTCL FLAT
  EXPT AUTO
  PIMX 800
  PIMN 100
  PTEV 0.0
  RATE .empty.
  SROT 16.0000
  EISE N
  EISA N/A
  HCTL Off
  AUPT N
  APTO OFF
  AUDO AUTO
  BROD .empty.
  BRID 0
  PVUL F
  PRJT GPRO
  SOFF 0
  CLKS 1
  CDAT 0x00000000639F9BE7
  SCTM 0
  PRNA 1
  PRNU 0
  SCAP N
  CDTM 0
  DUST NO_LIMIT
  VRES 3840, 3360
  VFPS 50000, 1000
  HSGT OFF
  BITR HIGH
  MMOD STEREO
  RAMP .empty.
  TZON 32767
  CLKC 0
  DZMX 1.4000
  CTRL Pro
  PWPR PERFORMANCE
  ORDP Y
  CLDP Y
  PIMD MANUAL
DEVC 
  DVID FOVL
  DVNM Large FOV
  ABSC 0.0000
  ZFOV 156.7126
  VFOV W
  MXCF x1
  MAPX 1.0000
  MYCF y1
  MAPY 1.0000
  PYCF r0r1r2r3r4r5r6
  POLY 0.0000, 2.1112, 0.1433, -1.0301, 0.6410, -0.0000, 0.0000
  ZMPL 0.7115
  ARUW 1.1429
  ARWA 1.1429
DEVC 
  DVID USRM
  DVNM User Metadata
  STRM 
    DBGL 1
    FRNT 1
    EXPS 1
    WAKE 2
    BOOT !Lspl
    USRD 1
DEVC 
  DVID HLMT
  DVNM Highlights

**Reply by dnewman-gpsw** [reply]
Something external to Labs is causing the captures to stop, and it likely displaying the error message on the LCD you don't have.

[2022-12-18 23:02:15] GP2 Temp.:57
[2022-12-18 23:02:18] GP2 Temp.:61
[2022-12-18 23:02:20] capture stopped:user input or error
[2022-12-18 23:02:20] capture stopped:user input or error
...

This means Labs found the capture stopped prematurely for a system reason: overheat, power, sd err etc.

The heartbeat errors suggest, camera is crashing.  It might be damaged somehow, overheating or power issues.  However the fact that many captures are working, suggest it there might be hope for the camera.  Still no lead on what is wrong.

**Reply by stpfl123** [reply]
_It might be damaged....... my 1 month old second naked 11 has the same issue
 somehow, overheating .... when error nr 3 temperature was only GP2 Temp.:40
 or power issues......... as i said it worked last half year without problems

**Comment by stpfl123** [comment]
Coud it be related to !MFRNT=1!MEXPS=1 ? 
Because i have some weired display errors ? 
Running PAL ? 
https://drive.google.com/file/d/13XDruqKnLxpep4uf5iQfW2uSJ0-tpOFn/view?usp=sharing, https://drive.google.com/file/d/1S7WjAhCP3gE6lL9OJrwjQPnuxeOkh9FS/view?usp=sharing

ISO Shutter text appears and disappear on front display like crazy when recording ? normal ? 
When not recording ISO Shutter text is static

**Reply by dnewman-gpsw** [reply]
Labs is experimental, so experiment with different settings.  I don't know your setup, so you will have to determine what works for you.  I have not heard of naked H11s having a problem in general.  My is working fine and I typically run a lot more Labs hacks (which should make things worse.) Although I shoot a more cinematic style so run 5k24 8:7 at 175Mb/s, for <3 minute captures (sub 250g rig.) I haven't done flight at 50fps, although sometimes 60.

P.S. Sometimes all you need is a factory reset.

**Reply by stpfl123** [reply]
without !MFRNT=1!MEXPS=1 it runs a lot smoother without display errors i will give it a try. 
thanks

**Reply by stpfl123** [reply]
i did some further research with the metadata tool.

before date 02.05.2023 i had firmwire H22.01.02.01.70
on date 02.05.2023 i made camera update to  H22.01.02.10.70

so the problem only occur with H22.01.02.10.70 !!!

**Reply by dnewman-gpsw** [reply]
So install the old version for now. It is available: https://github.com/gopro/labs/tree/master/docs/firmware/lfs

**Reply by stpfl123** [reply]
Thank you i already did and no errors so far with the old firmwire and !MFRNT=1!MEXPS=1
So for me the solution is 

either H22.01.02.10.70 without !MFRNT=1!MEXPS=1 command 
or H22.01.02.01.70 with !MFRNT=1!MEXPS=1

give me some days to proof that. 
Thank you for your help! You can close it i think its solved :-)

**Reply by stpfl123** [reply]
Is it possible to script not dipslay !MFRNT=1!MEXPS=1 when recording ?

**Reply by dnewman-gpsw** [reply]
Yes, with a script.  I don't think !MFRNT=1 is an issue on it own, as that doesn't add compute cycle unless there is something to display.   So you can enable and disable EXPS with a trigger script.   
```
<r0oMEXPS=1~oMEXPS=0!1N!R
```
